pkg/apis/servicecatalog/types.go

```diff
@@ -309,4 +309,15 @@ const (
 	BindingConditionReady BindingConditionType = "Ready"
```

Comment:
It feels like we're missing a state - either "ready" means created but not linked to the app yet, or it means "created and injected". Either way, there are two states at play but only one condition so I think we need another one.

A comment describing/outlining the flow between the states would help.